### PR TITLE
improve typing in multiple library actors

### LINKED ISSUE #476: error in pyright with strict mode for connect_robust
Hi,

We'd like to pass our project to strict mode with pyright however he's complaining about unknown key and values for `Dict` in `ssl_options`.
```
Type of "connect_robust" is partially unknown
  Type of "connect_robust" is "(url: str | URL = None, *, host: str = "localhost", port: int = 5672, login: str = "guest", password: str = "guest", virtualhost: str = "/", ssl: bool = False, loop: AbstractEventLoop = None, ssl_options: dict[Unknown, Unknown] = None, timeout: int | float | None = None, client_properties: Dict[str, bool | bytearray | Decimal | List[FieldValue] | FieldTable | float | int | str | datetime | None] = None, connection_class: Type[AbstractRobustConnection] = RobustConnection, **kwargs: Any) -> Coroutine[Any, Any, AbstractRobustConnection]"
```

I can make a PR to fix that if you want.

### PR DESCRIPTION
**Pull request by tvallois**
Closes #476 

I tried to fix as many errors as Pyright found but some of them are libraries related and cannot be fix in an easy way (need to fix libraries from the bottom)

### CODE REVIEW

**Comment by mosquito** [comment]
This project already formatted with gray formatter.

**Comment by mosquito** [comment]
This project was formatted with 79 chars maximum line length.

**Comment by mosquito** [comment]
@tvallois please resolve conflicts and rebase.

File: aio_pika/channel.py [hunk 1]
```diff
@@ -408,8 +400,7 @@ async def exchange_delete(
     def transaction(self) -> Transaction:
         if self.publisher_confirms:
             raise RuntimeError(
-                "Cannot create transaction when publisher "
-                "confirms are enabled",
+                "Cannot create transaction when publisher " "confirms are enabled",
```
Comment (decaz) [inline aio_pika/channel.py:403]: Unnecessary concatenation.

**Comment by mosquito** [comment]
Something went wrong on rebase stage. Changes from default branch is now changes in your PR.

**Comment by tvallois** [comment]
> Something went wrong on rebase stage. Changes from default branch is now changes in your PR.

Ok i'll rollback.

**Comment by tvallois** [comment]
> > Something went wrong on rebase stage. Changes from default branch is now changes in your PR.
> 
> Ok i'll rollback.

Ok i think i'm fine.

File: aio_pika/connection.py [hunk 2]
```diff
@@ -195,17 +195,17 @@ async def __aexit__(
...
     host: str = "localhost",
     port: int = 5672,
     login: str = "guest",
     password: str = "guest",
     virtualhost: str = "/",
     ssl: bool = False,
-    ssl_options: dict = None,
-    client_properties: FieldTable = None,
-    **kwargs: Any
+    ssl_options: Optional[Dict[str, Any]] = None,
```
Comment (mosquito) [inline aio_pika/connection.py:206]: The better solution is using TypedDict like this:

```python
class SSLOptions(TypedDict):
    cafile: Optional[str]
    capath: Optional[str]
    cadata: Optional[str]
    keyfile: Optional[str]
    certfile: Optional[str]
    no_verify_ssl: Optional[int]


def make_url(
     ...
     ssl_options: Optional[SSLOptions] = None,
     ...
):
    ...
```

Comment (mosquito) [reply aio_pika/connection.py:206]: The `SSLOptions` definition might be placed in abc.py

Comment (tvallois) [reply aio_pika/connection.py]: Done, perhaps it's better to use `total=False` and remove Optional values in the SSLOptions class?

File: aio_pika/abc.py [hunk 3]
```diff
@@ -32,6 +32,15 @@
 ZERO_TIME = datetime.utcfromtimestamp(0)
 
 
+class SSLOptions(TypedDict):
```
Comment (mosquito) [inline aio_pika/abc.py:35]: I guess this should be added to `__all__`

**Comment by mosquito** [comment]
@tvallois please run linter checks locally with `tox -e lint`, this will be faster than waiting my interaction with GH actions

**Comment by tvallois** [comment]
Hmm TypedDict does not exist in python version lower than 3.8. I can add `typing_extensions` dependency to resolve this but i'm not super fan to add dependencies just for this.

**Comment by mosquito** [comment]
@tvallois https://github.com/mosquito/aio-pika/runs/7562004324?check_suite_focus=true#step:5:24 TypedDict is not a part of python 3.7 stdlib. Sad but true.

**Comment by tvallois** [comment]
> @tvallois https://github.com/mosquito/aio-pika/runs/7562004324?check_suite_focus=true#step:5:24 TypedDict is not a part of python 3.7 stdlib. Sad but true.

Can replace with `Dict[str, Any]` or add `typing_extensions`as described above.

File: aio_pika/abc.py [hunk 4]
```diff
@@ -5,10 +5,16 @@
...
 from typing import (
-    Any, AsyncContextManager, AsyncIterable, Awaitable, Callable, Dict,
+    TYPE_CHECKING, Any, AsyncContextManager, AsyncIterable, Awaitable, Callable, Dict,
     FrozenSet, Generator, Iterator, MutableMapping, NamedTuple, Optional, Set,
     Tuple, Type, TypeVar, Union,
 )
+try:
+    from typing import TypedDict
+except ImportError:
+    if not TYPE_CHECKING:
```
Comment (mosquito) [inline aio_pika/abc.py:15]: I think you can not check it, just do the import

File: aio_pika/abc.py [hunk 5]
```diff
@@ -294,7 +309,7 @@ async def cancel(
     @abstractmethod
     async def get(
         self, *, no_ack: bool = False,
-        fail: bool = True, timeout: TimeoutType = 5,
+        fail: bool = True, timeout: TimeoutType = 5
```
Comment (mosquito) [inline aio_pika/abc.py:312]: And so, gray formatter and I really liked these commas.

File: aio_pika/abc.py (hunk 4 above)
Comment (tvallois) [reply aio_pika/abc.py:15]: pyright is complaining without it, cf: https://github.com/microsoft/pyright/issues/2193

**Review by mosquito — APPROVED** [review]

**Comment by mosquito** [comment]
@tvallois released in `aio-pika==8.1.1`

**Pull request merged**